### PR TITLE
cluster: flip envoy_reloadable_features_use_config_in_happy_eyeballs to true

### PR DESCRIPTION
**Pull request by fishpan1209**
Commit Message:  flip envoy_reloadable_features_use_config_in_happy_eyeballs to true
Additional Description: This flag has been flipped to true internally since March
Risk Level: low
Testing: tested internally for 4 month
Docs Changes:
Release Notes: Make flag envoy_reloadable_features_use_config_in_happy_eyeballs default to true
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by adisuissa — COMMENTED** [review]
Can you please add some context (how was it tested, for how long) in the PR description.

Also, this will probably require a release note.
/wait

**Comment by fishpan1209** [comment]
added description and release notes

**Comment by adisuissa** [comment]
> added description and release notes

I think the changes weren't pushed. can you please double check?

File: changelogs/current.yaml [hunk 1]
```diff
@@ -192,4 +192,9 @@ new_features:
     Added :ref:`delay_deny <envoy_v3_api_msg_extensions.filters.network.rbac.v3.RBAC>` to support deny connection after
     the configured duration.
 
+- area: cluster
```
Comment (adisuissa) [inline changelogs/current.yaml:195]: please remove blank line above this one.

Comment (adisuissa) [reply changelogs/current.yaml:195]: not sure why the blank line below before `deprecated:` was removed, and not the line above this one. Can you please take another look?

Comment (fishpan1209) [reply changelogs/current.yaml:194]: my mistake, fixed

**Review by adisuissa — DISMISSED** [review]
LGTM, thanks!

**Comment by adisuissa** [comment]
Ryan, can you please take a final pass on this (as the happy-eyeballs expert).
/assign @RyanTheOptimist

**Review by RyanTheOptimist — DISMISSED** [review]
Woo hoo!

**Comment by adisuissa** [comment]
Can you please merge main again, and hopefully it will fix the wedged CI test.

**Review by RyanTheOptimist — APPROVED** [review]

**Review by RyanTheOptimist — APPROVED** [review]

**Pull request merged**